bindings/java/java/src/main/java/org/ethereum/evmc/EvmcVm.java

```diff
@@ -7,37 +7,24 @@
 import static org.ethereum.evmc.Host.removeContext;
 
 import java.nio.ByteBuffer;
-import java.util.Objects;
 
 /**
  * The Java interface to the evm instance.
  *
  * <p>Defines the Java methods capable of accessing the evm implementation.
  */
 public final class EvmcVm implements AutoCloseable {
-  private static EvmcVm evmcVm;
-  private static boolean isEvmcLibraryLoaded = false;
   private ByteBuffer nativeVm;
+
   /**
    * This method loads the specified evm shared library and loads/initializes the jni bindings.
    *
+   * @param evmcPath the path to the evmc shared library
    * @param filename /path/filename of the evm shared object
    */
-  public static EvmcVm create(String filename) {
-    if (!EvmcVm.isEvmcLibraryLoaded) {
-      try {
-        // load so containing the jni bindings to evmc
-        System.load(System.getProperty("user.dir") + "/../c/build/lib/libevmc-java.so");
-        EvmcVm.isEvmcLibraryLoaded = true;
-      } catch (UnsatisfiedLinkError e) {
-        System.err.println("Native code library failed to load.\n" + e);
-        System.exit(1);
-      }
-    }
-    if (Objects.isNull(evmcVm)) {
-      evmcVm = new EvmcVm(filename);
-    }
-    return evmcVm;
+  public static EvmcVm create(String evmcPath, String filename) {
```

Comment:
Why aren't we following the old pattern of calling load once statically?
```
public final class EvmcVm {
   static {
      System.loadLibrary("libevmc.so");
   }
```

Reply:
Say you do this - this means the static block runs at the time the class is loaded.
That means less control as to when you perform the load. If the static block throws, the class can never be loaded.
A behavior I have seen in Java is this:
https://www.adamh.cz/blog/2012/12/how-to-load-native-jni-library-from-jar/
Basically, your java jar ships with the so/dylib/dll file embedded. On startup, it copies the file out to a temp directory. Then it loads from it.

Reply:
Exactly that seems to be the best approach. `libemvc-java.so` is not something the user should supply or load multiple times, it is the binding.